src/erlcloud_ec2.erl

```diff
@@ -2936,6 +2946,221 @@ net_if_params(#ec2_net_if{}=X) ->
+    describe_flow_logs(FlowIDs, [], Config);
+describe_flow_logs(MaxResults, Config)
+    when is_integer(MaxResults) andalso MaxResults >= ?FLOWS_MR_MIN andalso MaxResults =< ?FLOWS_MR_MAX,
+    is_record(Config, aws_config) ->
+    describe_flow_logs([], MaxResults, Config);
+describe_flow_logs(MaxResults, NextToken)
+    when is_integer(MaxResults) andalso MaxResults >= ?FLOWS_MR_MIN andalso MaxResults =< ?FLOWS_MR_MAX,
+    is_list(NextToken) ->
+    describe_flow_logs(MaxResults, NextToken, default_config()).
+
+-spec describe_flow_logs(ec2_flow_ids(), filter_list(), aws_config()) -> ok_error(proplist());
+                        (ec2_flow_ids(), filter_list(), ec2_max_result()) -> ok_error(proplist());
+                        (filter_list(), ec2_max_result(), aws_config()) -> ok_error(proplist(), ec2_token());
+                        (ec2_max_result(), ec2_token(), aws_config()) -> ok_error(proplist(), ec2_token()).
+describe_flow_logs(FlowIDs, Filter, Config)
+    when is_list(FlowIDs), is_list(Filter) orelse Filter =:= none , is_record(Config, aws_config) ->
+    Params = erlcloud_aws:param_list(FlowIDs, "FlowLogId") ++ list_to_ec2_filter(Filter),
+    case ec2_query(Config, "DescribeFlowLogs", Params, ?NEW_API_VERSION) of
+        {ok, Doc} ->
+            Flows = extract_results("DescribeFlowLogsResponse", "flowLogSet", fun extract_flow/1, Doc),
+            {ok, Flows};
+        {error, _} = E -> E
+    end;
+describe_flow_logs(none, MaxResults, Config)
+    when is_integer(MaxResults) andalso MaxResults >= ?FLOWS_MR_MIN andalso MaxResults =< ?FLOWS_MR_MAX,
+         is_record(Config, aws_config) ->
+    describe_flow_logs([], [], MaxResults, Config);
+describe_flow_logs([] = Filter, MaxResults, Config)
+    when is_integer(MaxResults) andalso MaxResults >= ?FLOWS_MR_MIN andalso MaxResults =< ?FLOWS_MR_MAX,
+         is_record(Config, aws_config) ->
+    describe_flow_logs([], Filter, MaxResults, Config);
+describe_flow_logs([{_, _} | _] = Filter, MaxResults, Config)
+    when is_integer(MaxResults) andalso MaxResults >= ?FLOWS_MR_MIN andalso MaxResults =< ?FLOWS_MR_MAX,
+         is_record(Config, aws_config) ->
+    describe_flow_logs([], Filter, MaxResults, Config);
+describe_flow_logs(MaxResults, NextToken, Config)
+    when is_integer(MaxResults),
+         is_list(NextToken),
+         is_record(Config, aws_config) ->
+    Params = [{"MaxResults", MaxResults}, {"NextToken", NextToken}],
+    case ec2_query(Config, "DescribeFlowLogs", Params, ?NEW_API_VERSION) of
+        {ok, Doc} ->
+            Flows = extract_results("DescribeFlowLogsResponse", "flowLogSet", fun extract_flow/1, Doc),
+            NewNextToken = extract_next_token("DescribeFlowLogsResponse", Doc),
+            {ok, Flows, NewNextToken};
+        {error,  _} = E -> E
+    end.
+
+-spec describe_flow_logs(ec2_flow_ids(), filter_list(), ec2_max_result(), aws_config()) -> ok_error(proplist(), ec2_token()).
+describe_flow_logs(FlowIDs, Filter, MaxResults, Config)
```

Comment:
According to [AWS API](http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeFlowLogs.html), `FlowIDs` and `MaxResults` cannot be specified in the same request, so erlcloud should not have an interface for that. 

Instead I recommend you implement the interface `(Filter, MaxResults, NextToken, Config)` so that pagination can be used by making a call `(Filter, MaxResult, undefined, Config)` and making consequent calls using `(Filter, MaxResults, NextToken, Config)` where `NextToken` is the token retrieved from previous calls. Same as it's implemented in `describe_instances/4`.